libslide/src/bignum.rs

```diff
@@ -0,0 +1,107 @@
+use std::char::from_digit;
+use std::fmt;
+
+// this probably can go in utils but I put it in here for now
+trait StringUtils {
+    fn substring(&self, start: usize, len: usize) -> Self;
+}
+
+impl StringUtils for String {
+    fn substring(&self, start: usize, len: usize) -> Self {
+        self.chars().skip(start).take(len).collect()
+    }
+}
+
+pub fn num(input: String) -> Bignum {
+    let bg = Bignum::new(input);
+    bg
+}
+
+pub struct Bignum {
+    // False is positive, True is negative
+    neg_sign: bool,
+    // we use vector since it will allow faster computation time
+    // for smaller numbers
+    int: Vec<u32>,
+    dec: Vec<u32>,
+}
+
+impl Bignum {
+    pub fn new(input: String) -> Bignum {
+        let neg_sign: bool;
+        let s: String;
+        let mut int: Vec<u32> = Vec::new();
+        let mut dec: Vec<u32> = Vec::new();
+        if input.chars().next().unwrap() == '-' {
+            neg_sign = true;
+            s = input.substring(1, input.chars().count() - 1);
+        } else {
+            neg_sign = false;
+            s = input;
+        }
+        let mut is_decimal = false;
+        for c in s.chars() {
+            match c {
+                x if x.is_digit(10) && is_decimal == false => int.push(c.to_digit(10).unwrap()),
+                x if x.is_digit(10) => dec.push(c.to_digit(10).unwrap()),
+                x if x == '.' => match is_decimal {
+                    true => panic!("string cannot have two decimal points!"),
+                    false => is_decimal = true,
+                },
+                _ => panic!("String contains not decimal values"),
+            }
+        }
+        Bignum {
+            neg_sign: neg_sign,
+            int: int,
+            dec: dec,
+        }
+    }
+}
+
+impl fmt::Display for Bignum {
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        let mut result: String = "".to_string();
+        if self.neg_sign {
+            result.push('-');
+        }
+        for i in 0..self.int.iter().count() {
+            result.push(from_digit(self.int[i], 10).unwrap());
+        }
+        if self.dec.iter().count() != 0 {
```

Comment:
This consumes the entire iterator. It's better to do something like
```suggestion
        if self.dec.len() > 0 {
```